core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -1655,6 +1663,217 @@ class KafkaApis(val requestChannel: RequestChannel,
     }
   }
 
+  def handleDescribeAcls(request: RequestChannel.Request): Unit = {
+    authorizeClusterAction(request)
+    val describeAclsRequest = request.body[DescribeAclsRequest]
+    authorizer match {
+      case None =>
+        def createResponse(throttleTimeMs: Int): AbstractResponse =
+          new DescribeAclsResponse(throttleTimeMs, new SecurityDisabledException(
+            "No Authorizer is configured on the broker."), Collections.emptySet[AclBinding]);
+        sendResponseMaybeThrottle(request, createResponse)
+      case Some(auth) =>
+        val filter = describeAclsRequest.filter()
+        var returnedAcls = new util.ArrayList[AclBinding]
+        val aclMap : Map[Resource, Set[Acl]] = auth.getAcls()
+        aclMap.foreach {
+          case (resource, acls) => {
+            acls.foreach {
+              case (acl) => {
+                val fixture = new AclBinding(new AdminResource(AdminResourceType.fromString(resource.resourceType.toString), resource.name),
+                    new AccessControlEntry(acl.principal.toString(), acl.host.toString(), acl.operation.toJava, acl.permissionType.toJava))
+                if (filter.matches(fixture))
+                  returnedAcls.add(fixture)
+              }
+            }
+          }
+        }
+        def createResponse(throttleTimeMs: Int): AbstractResponse =
+          new DescribeAclsResponse(throttleTimeMs, null, returnedAcls)
+        sendResponseMaybeThrottle(request, createResponse)
```

Comment:
Isn't it easier to just inline `createResponse`? eg:

```scala
sendResponseMaybeThrottle(request, throttleTimeMs => new DescribeAclsResponse(throttleTimeMs, null, returnedAcls))
```
A few others are the same.